test/extended/util/disruption/controlplane/known_backends.go

```diff
@@ -210,6 +280,78 @@ func createOAuthAPIMonitoringWithConnectionReuseAgainstAPICache(clusterConfig *r
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "kube-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.NewConnectionType,
+			Protocol:         builder.ProtocolHTTP2,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
+
+func createKubeAPIMonitoringWithConnectionReuseHTTP2(clusterConfig *rest.Config, b *builder.CIDisruptionTestBuilder) (*builder.LegacyBackendSampler, error) {
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "kube-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.ReusedConnectionType,
+			Protocol:         builder.ProtocolHTTP2,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
+
+func createKubeAPIMonitoringWithNewConnectionsHTTP1(clusterConfig *rest.Config, b *builder.CIDisruptionTestBuilder) (*builder.LegacyBackendSampler, error) {
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "kube-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.NewConnectionType,
+			Protocol:         builder.ProtocolHTTP1,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
+
+func createKubeAPIMonitoringWithConnectionReuseHTTP1(clusterConfig *rest.Config, b *builder.CIDisruptionTestBuilder) (*builder.LegacyBackendSampler, error) {
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "kube-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.ReusedConnectionType,
+			Protocol:         builder.ProtocolHTTP1,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
+
+func createOpenShiftAPIMonitoringWithNewConnectionsHTTP2(clusterConfig *rest.Config, b *builder.CIDisruptionTestBuilder) (*builder.LegacyBackendSampler, error) {
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "openshift-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.NewConnectionType,
+			Protocol:         builder.ProtocolHTTP2,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
+
+func createOpenShiftAPIMonitoringWithConnectionReuseHTTP2(clusterConfig *rest.Config, b *builder.CIDisruptionTestBuilder) (*builder.LegacyBackendSampler, error) {
+	return b.Build(builder.CIDisruptionTest{
+		DisruptionTypeParams: builder.DisruptionTypeParams{
+			TargetServer:     "openshift-apiserver",
+			LoadBalancerType: builder.ExternalLoadBalancerType,
+			ConnectionType:   monitorapi.ReusedConnectionType,
+			Protocol:         builder.ProtocolHTTP2,
+		},
+		Path: "/api/v1/namespaces/default",
+	})
+}
```

Comment:
Via: https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/27838/pull-ci-openshift-origin-master-e2e-aws-ovn-upgrade/1663323849552302080/artifacts/e2e-aws-ovn-upgrade/openshift-e2e-test/artifacts/junit/backend-disruption_20230530-000958.json

Output still not quite identical with the old backends.

Old style:
```json
        "ingress-to-console-reused-connections": {
            "Name": "ingress-to-console-reused-connections",
            "BackendName": "ingress-to-console",
            "ConnectionType": "Reused",
            "DisruptedDuration": "0s",
            "DisruptionMessages": null,
            "LoadBalancerType": "external-lb",
            "Protocol": "http1",
            "TargetAPI": ""
        },
```

New style:
```json
        "kube-api-http2-external-lb-new": {
            "Name": "kube-api-http2-external-lb-new",
            "BackendName": "kube-api-http2-external-lb-new",
            "ConnectionType": "New",
            "DisruptedDuration": "0s",
            "DisruptionMessages": null,
            "LoadBalancerType": "external-lb",
            "Protocol": "http2",
            "TargetAPI": "kube-api"
        },
```

Specifically to match, the new json should be:
```json
        "kube-api-http2-external-lb-new-connections": {
            "Name": "kube-api-http2-external-lb-new-connections",
            "BackendName": "kube-api-http2-external-lb",
            "ConnectionType": "New",
            "DisruptedDuration": "0s",
            "DisruptionMessages": null,
            "LoadBalancerType": "external-lb",
            "Protocol": "http2",
            "TargetAPI": "kube-api"
        },
```

Specifically: Add "-connections" to the key in the map, and the BackendName, drop "-new" from the Name.

I prefer the way you did it, but I'm worried someone out there is assuming -connections at the end, or using that Name. Staying consistent will keep us out of trouble with messing up data in the db in ways I'd rather not have to go clean up.

If you could get TargetAPI populated in the old backends, we could potentially start looking at using this data somewhere instead of assuming unique BackendNames that encompass all the details.

It's nice to see every piece broken out explicitly in it's own field, we should make use of that eventually.